### PR TITLE
[incubator/azuremonitor-containers] add additional parameters and update the instructions for Helm3

### PR DESCRIPTION
**Pull request by ganga1980**
#### What this PR does / why we need it:
This PR has the following updates
1. update the instructions for Helm3  client
2. add additional parameters clusterId and clusterRegion so that same chart can be used for the managed clusters hosted in Azure
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: incubator/azuremonitor-containers/templates/NOTES.txt [hunk 1]
```diff
@@ -14,21 +14,37 @@
...
    --set omsagent.secret.wsid=<your_workspace_id> \
    --set omsagent.secret.key=<your_workspace_key> \
    --set omsagent.env.clusterName=<your_cluster_name> \
+   incubator/azuremonitor-containers
+
+       or
+  helm upgrade {{ .Release.Name }} \
+   --set omsagent.secret.wsid=<your_workspace_id> \
+   --set omsagent.secret.key=<your_workspace_key> \
+   --set omsagent.env.clusterId=<your_cluster_id> \
```
Comment (vishiy) [inline incubator/azuremonitor-containers/templates/NOTES.txt]: can we please not document this explicitly in notes ? Hidden params does not need to be in notes

File: incubator/azuremonitor-containers/templates/omsagent-daemonset.yaml [hunk 2]
```diff
@@ -34,8 +34,15 @@ spec:
        resources:
 {{ toYaml .Values.omsagent.resources.daemonset | indent 9 }}
        env:
+       {{- if ne .Values.omsagent.env.clusterId "<your_cluster_id>" }}
+       - name: AKS_RESOURCE_ID
+         value: {{ .Values.omsagent.env.clusterId | quote }}
+       - name: AKS_REGION
```
Comment (vishiy) [inline incubator/azuremonitor-containers/templates/omsagent-daemonset.yaml:41]: better to check explicitly for region being present (just like clusterid)

File: incubator/azuremonitor-containers/templates/omsagent-deployment.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-{{- if and (ne .Values.omsagent.secret.key "<your_workspace_key>") (ne .Values.omsagent.secret.wsid "<your_workspace_id>") (ne .Values.omsagent.env.clusterName "<your_cluster_name>")}}
+{{- if and (ne .Values.omsagent.secret.key "<your_workspace_key>") (ne .Values.omsagent.secret.wsid "<your_workspace_id>") (or (ne .Values.omsagent.env.clusterName "<your_cluster_name>") (ne .Values.omsagent.env.clusterId "<your_cluster_id>"))}}
```
Comment (vishiy) [inline incubator/azuremonitor-containers/templates/omsagent-deployment.yaml:1]: please check if all combinations are working by testing (with/without clusterid and/or region)

File: incubator/azuremonitor-containers/templates/omsagent-deployment.yaml [hunk 4]
```diff
@@ -38,8 +38,15 @@ spec:
        resources:
 {{ toYaml .Values.omsagent.resources.deployment | indent 9 }}
        env:
+       {{- if ne .Values.omsagent.env.clusterId "<your_cluster_id>" }}
+       - name: AKS_RESOURCE_ID
+         value: {{ .Values.omsagent.env.clusterId | quote }}
+       - name: AKS_REGION
+         value: {{ .Values.omsagent.env.clusterRegion | quote }}
```
Comment (vishiy) [inline incubator/azuremonitor-containers/templates/omsagent-deployment.yaml:46]: same. better to check for region being present (like clusterid)

File: incubator/azuremonitor-containers/README.md [hunk 5]
```diff
@@ -29,11 +31,12 @@ This article describes how to set up and use [Azure Monitor - Containers](https:
 
 ### To Use Azure Log Analytics Workspace in Public Cloud
 
+> Note: If you are using Helm3 client, use `--generate-name` instead of `--name myrelease-1` for the install command
```
Comment (lachie83) [inline incubator/azuremonitor-containers/README.md]: This isn't necessary

File: incubator/azuremonitor-containers/README.md [hunk 6]
```diff
@@ -55,7 +58,8 @@ $ helm install --name myrelease-1 \
 
 ## Uninstalling the Chart
 
-To uninstall/delete the `myrelease-1` release:
+To uninstall/delete the installed azure monitor for container release:
```
Comment (lachie83) [inline incubator/azuremonitor-containers/README.md]: Also this isn't necessary

File: incubator/azuremonitor-containers/templates/omsagent-deployment.yaml (hunk 3 above)
Comment (ganga1980) [reply incubator/azuremonitor-containers/templates/omsagent-deployment.yaml:1]: > please check if all combinations are working by testing (with/without clusterid and/or region)
> yep, tested.

File: incubator/azuremonitor-containers/templates/omsagent-daemonset.yaml (hunk 2 above)
Comment (ganga1980) [reply incubator/azuremonitor-containers/templates/omsagent-daemonset.yaml:41]: > better to check explicitly for region being present (just like clusterid)
> sure, will add that.

File: incubator/azuremonitor-containers/README.md (hunk 6 above)
Comment (ganga1980) [reply incubator/azuremonitor-containers/README.md]: > Also this isn't necessary

Update the note to cover deprecated flags in helm3

File: incubator/azuremonitor-containers/templates/NOTES.txt (hunk 1 above)
Comment (ganga1980) [reply incubator/azuremonitor-containers/templates/NOTES.txt]: > can we please not document this explicitly in notes ? Hidden params does not need to be in notes
> fixed as discussed.

File: incubator/azuremonitor-containers/templates/omsagent-daemonset.yaml (hunk 2 above)
Comment (ganga1980) [reply incubator/azuremonitor-containers/templates/omsagent-daemonset.yaml:41]: > > better to check explicitly for region being present (just like clusterid)
> > sure, will add that.

>> fixed.

File: incubator/azuremonitor-containers/templates/omsagent-deployment.yaml (hunk 4 above)
Comment (ganga1980) [reply incubator/azuremonitor-containers/templates/omsagent-deployment.yaml:46]: > same. better to check for region being present (like clusterid)

Fixed.

File: incubator/azuremonitor-containers/README.md (hunk 5 above)
Comment (ganga1980) [reply incubator/azuremonitor-containers/README.md]: > This isn't necessary

revised Note to cover the deprecated flag.

**Comment by lachie83** [comment]
@vishiy are you good with this?

**Review by vishiy — APPROVED** [review]

**Pull request merged**